src/util/yaml.ts

```diff
@@ -0,0 +1,164 @@
+import { Scalar, YAMLMap, YAMLSeq } from "yaml/types";
+import { PPAbleError } from "./errors";
+import { Location, Range } from "./location";
+import { MacroFile } from "./sources";
+import YAML from "yaml";
+import { single } from "./misc";
+import { assert } from "solc-typed-ast";
+
+const srcloc = require("src-location");
+
+export class YamlSchemaError extends PPAbleError {}
+
+function makeYamlLoc(off: number, file: MacroFile): Location {
+    const t = srcloc.indexToLocation(file.contents, off);
+    const { line, column } = t;
+    return {
+        offset: off,
+        line,
+        column,
+        file
+    };
+}
+
+/// Turn a YAML range into our internal range. Note that yaml ranges are [start, end], not [start, len]
+export function makeYamlRange(range: [number, number], file: MacroFile): Range {
+    return { start: makeYamlLoc(range[0], file), end: makeYamlLoc(range[1], file) };
+}
+
+/**
+ * Check that a parsed YAML `node` adheres to the given `schema`. We specify `schema` as a JSON object. For example the below object:
+ * {
+ *  "a": "string",
+ *  "*": {
+ *      "b": "number",
+ *      "c": [{
+ *          "d": "string"
+ *      }]
+ *  }
+ * }
+ *
+ * Specifies that:
+ * 1. The top-level yaml node should be a map
+ * 2. The top-level yaml node should have at least 1 entry "a" which is a scalar string
+ * 3. Any other entries in the top-level yaml node must be yaml maps too
+ * 4. Objects for other entries in the top-level yaml node must have two keys - "b" and "c"
+ * 5. Values for the "b" key must be scalar numbers
+ * 6. Values for the "c" key must be sequences, containing maps with a single key "d" which is a scalar string.
+ * @param node
+ * @param schema
+ * @param file
```

Comment:
Params are not described, while self explaindatory. Maybe remove them?

Reply:
Fixed in 8ae6a66.